ckan_sdk/client.py

```diff
@@ -3,92 +3,99 @@
-
-        # Get the JWT token from CkanAuthz
-        scope = 'obj:{}/{}/*:write'.format(self.organization_id, self.dataset_id)
-        response = self.get_jwt_from_ckan_authz(scope)
-        jwt_token = response['result']['token']
-
-        file_info = self.get_file_info(resource_path)
-        lfs_response = self.request_file_upload_actions(jwt_token,
-                             file_info.get('sha256'), file_info.get('size'))
-
-        lfs_object = lfs_response.get('objects')[0]
-        result = {
-            'oid': lfs_object['oid'],
-            'size': lfs_object['size'],
-            'name': file_info.get('name'),
-            'success': True,
-            'file_already_exists': False,
-            }
 
-        # File doesn't exist in storage
-        if lfs_object['actions']:
+    def push(self, dataset, append=False):
+        # Push resources to CKAN and cloud
 
-            # Upload the file to cloud storage
-            actions = lfs_object['actions']
-            upload_url = actions['upload']['href']
-            upload_token = actions['upload']['header']
-            self.upload_to_storage(upload_url, upload_token, file_info.get('content'), on_progress)
-
-            # Verify file to cloud storage
-            verify_url = actions['verify']['href']
-            verify_token = actions['verify']['header']
-            self.verify_upload(verify_url, verify_token, file_info.get('sha256'), file_info.get('size'))
-
-            return result
+        try:
+            resources = dataset.pop('resources')
+        except KeyError:
+            log.exception('There is no resource in the dataset {}'.format(dataset.get('name')))
+            return
 
-        # File is already in storage
+        if not append:
+            dataset['owner_org'] = self.organization_id
+            dataset = self._ckan_package_or_resource_create(dataset, 'api/3/action/package_create')['result']
         else:
-            result['file_already_exists'] == True
-
-    def push_resources(self, list_resources_path=None, resources_directory_path=None):
-        # Upload multiple resources (you can either provide a list of resource or a path to the directory)
-
-        result = []
-        if list_resources_path:
-            for resource_path in list_resources_path:
-                result.append(self.push(resource_path))
-
-        elif resources_directory_path:
-            resources = self.find_csv_filenames(resources_directory_path)
-            for resource in resources:
-                result.append(self.push(os.path.join(resources_directory_path, resource)))
-
-        return result
-
-    def push_dataset_metadata(self):
-        pass
-
-    def push_resource_metadata(self):
-        pass
+            resources = [resources[-1]]
+
+        res = []
+        for resource in resources:
+            resource['package_id'] = dataset['id']
+            resource = self._ckan_package_or_resource_create(resource, 'api/3/action/resource_create')['result']
```

Comment:
don't you want to create the resource *after* you have pushed to cloud storage? Then if cloud storage upload fails in some way you haven't created an object in CKAN without file being upload ...